support/runtime-upgrade/index.js

```diff
@@ -0,0 +1,57 @@
+const { ApiPromise, WsProvider, Keyring } = require('@polkadot/api');
+
+var argv = require('minimist')(process.argv.slice(2));
+const fs = require('fs');
+
+// Create a promise API instance of the passed in node address.
+async function createPromiseApi(nodeAddress) {
+    const wsProvider = new WsProvider(nodeAddress);
+    const api = await new ApiPromise({ provider: wsProvider });
+    await api.isReady;
+    return api;
+}
+
+async function main() {
+    const nodeAddress = argv.nodeAddress;
+    const api = await createPromiseApi(nodeAddress);
+    const keyring = new Keyring({ type: 'sr25519' });
+    // Retrieve the upgrade key from the chain state
+    const adminId = await api.query.sudo.key();
```

Comment:
This is declared pretty far away from where it is used (logically speaking). I'd recommend declaring things as late a possible for a few reasons:

1. Limits what readers need to think about.
2. Communicates the scope of the relevance of the variable better.
3. Keeps failures/messages more in context. For example, if we couldn't complete the RPC and you specified a bad path for the mnemonic, we would fail first with the RPC, when we didn't actually need to yet.